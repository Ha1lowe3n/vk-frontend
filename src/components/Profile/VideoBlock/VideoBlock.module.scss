.videoBlock {
    .videos {
        padding: 12px 15px 15px;
    }
}
