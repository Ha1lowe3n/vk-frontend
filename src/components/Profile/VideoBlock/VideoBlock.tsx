import React from "react";

import styles from "./VideoBlock.module.scss";

import { WhiteBlockInfo } from "../../WhiteBlock/WhiteBlockInfo/WhiteBlockInfo";
import { Video } from "./Video/Video";

export const VideoBlock: React.FC = () => {
    return (
        <div className={styles.videoBlock}>
            <WhiteBlockInfo title={"Videos"} count={208} />
            <div className={styles.videos}>
                <Video />
                <Video />
            </div>
        </div>
    );
};
