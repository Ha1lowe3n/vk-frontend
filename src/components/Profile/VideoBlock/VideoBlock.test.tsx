import { fireEvent, render, screen } from "@testing-library/react";
import { VideoBlock } from "./VideoBlock";

it("render VideoBlock component", () => {
    render(<VideoBlock />);
    expect(screen.getByText("Videos")).toBeInTheDocument();
    expect(screen.getByText("208")).toBeInTheDocument();
});

it("VideoBlock snapshot", () => {
    const view = render(<VideoBlock />);
    expect(view).toMatchSnapshot();
});
