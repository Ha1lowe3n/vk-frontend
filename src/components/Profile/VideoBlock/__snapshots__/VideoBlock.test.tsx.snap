// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoBlock snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="videoBlock"
      >
        <div
          class="whiteBlockInfo"
        >
          <span
            class="title"
          >
            Videos
          </span>
          <span
            class="count"
          >
            208
          </span>
        </div>
        <div
          class="videos"
        >
          <div
            class="video"
          >
            <a
              href="#"
            >
              <div
                class="videoImg"
                style="background-image: url(https://i.imgur.com/FDvVliP.jpg);"
              >
                <div
                  class="videoPlayIcon"
                  data-testid="videoPlayIcon"
                />
              </div>
            </a>
            <a
              class="link_text videoTitle"
              href="#"
            >
              Вот что делают монстры под кроватью
            </a>
          </div>
          <div
            class="video"
          >
            <a
              href="#"
            >
              <div
                class="videoImg"
                style="background-image: url(https://i.imgur.com/FDvVliP.jpg);"
              >
                <div
                  class="videoPlayIcon"
                  data-testid="videoPlayIcon"
                />
              </div>
            </a>
            <a
              class="link_text videoTitle"
              href="#"
            >
              Вот что делают монстры под кроватью
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="videoBlock"
    >
      <div
        class="whiteBlockInfo"
      >
        <span
          class="title"
        >
          Videos
        </span>
        <span
          class="count"
        >
          208
        </span>
      </div>
      <div
        class="videos"
      >
        <div
          class="video"
        >
          <a
            href="#"
          >
            <div
              class="videoImg"
              style="background-image: url(https://i.imgur.com/FDvVliP.jpg);"
            >
              <div
                class="videoPlayIcon"
                data-testid="videoPlayIcon"
              />
            </div>
          </a>
          <a
            class="link_text videoTitle"
            href="#"
          >
            Вот что делают монстры под кроватью
          </a>
        </div>
        <div
          class="video"
        >
          <a
            href="#"
          >
            <div
              class="videoImg"
              style="background-image: url(https://i.imgur.com/FDvVliP.jpg);"
            >
              <div
                class="videoPlayIcon"
                data-testid="videoPlayIcon"
              />
            </div>
          </a>
          <a
            class="link_text videoTitle"
            href="#"
          >
            Вот что делают монстры под кроватью
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
